integrationtests/self/http_test.go

```diff
@@ -528,4 +528,30 @@ var _ = Describe("HTTP tests", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(resp.StatusCode).To(Equal(200))
 	})
+
+	It("sets conn context", func() {
+		type ctxKey int
+		server.ConnContext = func(ctx context.Context, c quic.Connection) context.Context {
+			ctx = context.WithValue(ctx, ctxKey(0), "Hello")
+			ctx = context.WithValue(ctx, ctxKey(1), c)
+			return ctx
+		}
+		mux.HandleFunc("/conn-context", func(w http.ResponseWriter, r *http.Request) {
+			defer GinkgoRecover()
+			v, ok := r.Context().Value(ctxKey(0)).(string)
+			Expect(ok).To(BeTrue())
+			Expect(v).To(Equal("Hello"))
+
+			_, ok = r.Context().Value(ctxKey(1)).(quic.Connection)
+			Expect(ok).ToNot(Equal(nil))
+
+			serv, ok := r.Context().Value(http3.ServerContextKey).(*http3.Server)
```

Comment:
This check should be in the `ConnContext` function, right?

Reply:
Sure. How about we put it in both the `ConnContext` function and the handler?

Reply:
Sure, that works as well.